### PR TITLE
Feature/payouts different accounts

### PR DESCRIPTION
**Pull request by vonElfvin**
# Jira Issue: [HVG-841]

## What?
- Use different payout accounts for different carriers when making claim payouts

## Why?
- So we can have members with claims for HDI/HEdvig/Other as a carrier

## Optional checklist
- [x] Codecouted
- [x] Unit tests written
- [ ] Tested locally

[HVG-841]: https://hedvig.atlassian.net/browse/HVG-841

### CODE REVIEW

File: payment-service/src/main/java/com/hedvig/paymentservice/configuration/Trustly.java [hunk 1]
```diff
@@ -17,39 +15,54 @@
...
 
-  @Value("${hedvig.trustly.claim.username}")
-  String usernameClaim;
+    @Value("${hedvig.trustly.claimx.username}")
+    String usernameClaimX;
 
-  @Value("${hedvig.trustly.claim.password}")
-  String passwordClaim;
+    @Value("${hedvig.trustly.claimx.password}")
+    String passwordClaimX;
```
Comment (meletios) [inline payment-service/src/main/java/com/hedvig/paymentservice/configuration/Trustly.java:40]: Not sure if claimx is very intuitive for a developer that doesn't know the back story. 
Shall we rename the claim credential to 

trustly.claim.hedvig.*
trustly.claim.hdi.*

I think it makes more sense right?

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt [hunk 2]
```diff
@@ -506,6 +512,7 @@ class Member() {
 
     private fun getSingleTransaction(
         transactionId: UUID
+
```
Comment (meletios) [inline payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt:515]: extra space?

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/upcasters/PayoutCreatedEventUpCaster.java [hunk 3]
```diff
@@ -28,9 +28,6 @@ protected IntermediateEventRepresentation doUpcast(IntermediateEventRepresentati
         document -> {
           Element element = document.getRootElement();
           element.addElement("category").setText(TransactionCategory.CLAIM.name());
-          element.addElement("referenceId");
-          element.addElement("note");
-          element.addElement("handler");
```
Comment (meletios) [inline payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/upcasters/PayoutCreatedEventUpCaster.java:33]: Do we care about the previous upcaster? i guess we should not change that, right?

File: trustly-client/src/main/java/com/hedvig/paymentService/trustly/Account.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 public enum Account {
   PREMIUM,
   CLAIM,
+  CLAIM_X,
```
Comment (meletios) [inline trustly-client/src/main/java/com/hedvig/paymentService/trustly/Account.java:6]:
```suggestion
  CLAIM_HDI,
  CLAIM_HEDVIG,
```

File: trustly-client/src/main/java/com/hedvig/paymentService/trustly/security/SignatureHandler.java [hunk 5]
```diff
@@ -85,12 +87,17 @@ public void init(
     final String passwordPremium,
     final String usernameClaim,
     final String passwordClaim,
-    final boolean testEnvironment)
+    final String usernameClaimX,
+    final String passwordClaimX,
```
Comment (meletios) [inline trustly-client/src/main/java/com/hedvig/paymentService/trustly/security/SignatureHandler.java:91]:
```suggestion
    final String usernameClaimHedvig,
    final String passwordClaimHedvig,
```

**Review by meletios — CHANGES_REQUESTED** [review]
Sweet, i think it would be valuable to be more specific with the account names.

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/upcasters/PayoutCreatedEventUpCaster.java (hunk 3 above)
Comment (vonElfvin) [reply payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/upcasters/PayoutCreatedEventUpCaster.java:33]: Upcasts without values do nothing, thus this is more or less removing 3 lines that don't do anything :)

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt (hunk 2 above)
Comment (vonElfvin) [reply payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt]: Ktlint 🤷

File: payment-service/src/main/java/com/hedvig/paymentservice/configuration/Trustly.java (hunk 1 above)
Comment (vonElfvin) [reply payment-service/src/main/java/com/hedvig/paymentservice/configuration/Trustly.java]: I agree

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt [hunk 6]
```diff
@@ -172,6 +173,12 @@ class Member() {
         }
 
         adyenPayoutAccount?.let { account ->
+            require(command.category == TransactionCategory.CLAIM) {
+                throw IllegalArgumentException("Illegal to create payout with adyen that is not of type claim (memberId=$memberId)")
+            }
+            require(command.carrier == Carrier.HEDVIG) {
+                throw IllegalArgumentException("Illegal to create payout with adyen that does not have Hedvig as a carrier (memberId=$memberId)")
+            }
```
Comment (meletios) [inline payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt:181]: We should add a feature flag here as well. Just to not block the Swedish payout splitting. Or?

**Review by meletios — APPROVED** [review]
Nice, shall we add a feature flag for the Adyen payout?

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt (hunk 6 above)
Comment (vonElfvin) [reply payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Member.kt:181]: This is inside the adyen one so it should be fine 👍

**Pull request merged**